File: task1.py
import math
eps = 1
while 1 + eps / 2 > 1:
    eps /= 2

def pf(msg, val):
    print(msg, val, float.hex(val))

pf('eps =', eps)
for i in range(1, 10):
    pf(f'{i}/2', 1 + eps * (i // 2) + (eps / 2 if i % 2 else 0))

while eps / 2 > 0:
    eps /= 2
print('min =', float.hex(eps), eps)

while eps * 2 < math.inf:
    eps *= 2
print('max =', float.hex(eps), eps)

File: task1.c
#include <stdio.h>
#include <math.h>

int main(int argc, char **argv) {
    double a = 1;
    while (1 + a / 2 > 1)
        a /= 2;
    printf("eps = %.13a %e\n", a, a);

    while (a / 2 > 0)
        a /= 2;
    printf("min = %.13a %e\n", a, a);

    while (a * 2 < INFINITY)
        a *= 2;
    printf("max = %.13a %e\n", a, a);
    return 0;
}
